### ISSUE TITLE
Still a regression in withResizeDetector typing

### ISSUE THREAD

**Issue by grasshoppergn** [maslianok/react-resize-detector#165]
Hi! This is a duplicate of #150 with a wish to reopen it - I couldn't respond to that one in time, so the argument was probably missed.

**Comment by maslianok** [comment]
**We need someone's help with this issue.**

I spent a few hours trying to resolve it and I failed. I can't simply copy/paste the approach that worked earlier because we've changed some logic inside `withResizeDetector` HOC.

_For future reference._

The [previous approach](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/b5438e8c35c2b4ab1020fb88e5760117922eed60/types/react-resize-detector/index.d.ts#L103)

```jsx
interface ReactResizeDetectorDimensions {
    height: number;
    width: number;
}

export function withResizeDetector<T extends Partial<ReactResizeDetectorDimensions>>(
    WrappedComponent: React.ComponentType<T>,
    props?: ReactResizeDetectorProps,
): React.ComponentType<Omit<T, keyof ReactResizeDetectorDimensions>>;
```

The [current typings](https://github.com/maslianok/react-resize-detector/blob/master/src/withResizeDetector.tsx#L6):

```jsx
function withResizeDetector<P, ElementT extends HTMLElement = HTMLElement>(
  ComponentInner: ComponentType<P>,
  options: ComponentsProps<ElementT> = {}
)
```

Adding `ComponentType<Omit<T, keyof ReactResizeDetectorDimensions>>;` doesn't solve the problem - TS throws "Type is not assignable" error.

**Comment by maslianok** [comment]
Published in v6.7.4.

@grasshoppergn thanks for helping with this issue!

**Comment by grasshoppergn** [comment]
My pleasure :)
